test/concurrent-emit-fail-interrupt.cc

```diff
@@ -46,10 +46,12 @@ TYPED_TEST(ConcurrentTypedTest, EmitFailInterrupt) {
                })
         | this->ConcurrentOrConcurrentOrdered([&]() {
             return Map(Let([&](int& i) {
-              return Eventual<std::string>([&](auto& k) {
-                k.Fail(std::runtime_error("error"));
-                interrupt.Trigger();
-              });
+              return Eventual<std::string>()
```

Comment:
Let's make Eventual be configurated using different order (for example raises->interruptible->start, start->raises) because it depends and need to be sure that the future changes won't break it.

Reply:
There isn't anything preventing you from configuring in any order!